demos/noise_suppression_demo/cpp/main.cpp

```diff
@@ -2,67 +2,49 @@
-#include "utils/common.hpp"
-#include "utils/slog.hpp"
-
-typedef std::chrono::steady_clock Time;
-
-static const char help_message[] = "Print a usage message.";
-static const char inp_wav_message[] = "Required. Path to a input WAV file.";
-static const char out_wav_message[] = "Optional. Path to a output WAV file.";
-static const char model_message[] = "Required. Path to an .xml file with a trained model.";
-static const char target_device_message[] = "Optional. Specify the target device to infer on (the list of available "
-                                            "devices is shown below). Default value is CPU. "
-                                            "The demo will look for a suitable plugin for device specified.";
-DEFINE_bool(h, false, help_message);
-DEFINE_string(i, "", inp_wav_message);
-DEFINE_string(o, "noise_suppression_demo_out.wav", out_wav_message);
-DEFINE_string(m, "", model_message);
-DEFINE_string(d, "CPU", target_device_message);
-
-static void showUsage() {
-    std::cout << std::endl;
-    std::cout << "noise_suppression_demo [OPTION]" << std::endl;
-    std::cout << "Options:" << std::endl;
-    std::cout << std::endl;
-    std::cout << "    -h           " << help_message << std::endl;
-    std::cout << "    -i INPUT     " << inp_wav_message << std::endl;
-    std::cout << "    -o OUTPUT    " << out_wav_message << std::endl;
-    std::cout << "    -m MODEL     " << model_message << std::endl;
-    std::cout << "    -d DEVICE    " << target_device_message << std::endl;
-}
-
-bool ParseAndCheckCommandLine(int argc, char* argv[]) {
-    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
-    if (FLAGS_h) {
-        showUsage();
-        return false;
+#include <utils/common.hpp>
+#include <utils/slog.hpp>
+
+#include <gflags/gflags.h>
+#include <openvino/openvino.hpp>
+
+using namespace std;
+
+namespace {
+constexpr char h_msg[] = "show the [H]elp message and exit";
+DEFINE_bool(h, false, h_msg);
+
+constexpr char m_msg[] = "path to an .xml file with a trained [M]odel";
+DEFINE_string(m, "", m_msg);
+
+constexpr char i_msg[] = "path to an [I]nput 16kHz WAV file";
+DEFINE_string(i, "", i_msg);
+
+constexpr char d_msg[] = "specify a [D]evice to infer on (the list of available devices is shown below)";
+DEFINE_string(d, "CPU", d_msg);
+
+constexpr char o_msg[] = "path to an [O]utput WAV file";
+DEFINE_string(o, "noise_suppression_demo_out.wav", o_msg);
+
+void parse(int argc, char *argv[]) {
+    gflags::ParseCommandLineFlags(&argc, &argv, false);
+    slog::info << ov::get_openvino_version() << slog::endl;
+    if (FLAGS_h || 1 == argc) {
+        cout <<   "\t[-h]                                  " << h_msg
+             << "\n\t[--help]                              print [HELP] on all arguments"
+             << "\n\t -m <[M]odel file>                    " << m_msg
+             << "\n\t[-i] <[I]nput file>                   " << i_msg
+             << "\n\t[-d] <CPU>                            " << d_msg
```

Comment:
I know. And I follow this for required args. For optional args, I thought it would be nice to have an example of correct value. Also it reflects default value. So when you run python demo with short help `-h`, it would print not only an arg name, but also a default value

Reply:
It is sitll non-informative. If I see it first time, I'd think that I had to provide type of CPU or some CPU-related parameter here. I still strongly suggest "[-d] <device>" and provide information about default value in d_msg

Reply:
`-d <DEVICE>` is also strange because `d` already stands for device.

Reply:
<DEVICE> might be not perfect but at least it not limits to CPU only. Might be changed to <DEVICE_NAME> for example, but is longer without real need.